hallux/main.py

```diff
@@ -149,34 +149,6 @@ def get_version():
         return "DEVELOP"
 
 
-def validate_model_args(argv, model_index: int) -> str | None:
-    model_value = find_argvalue(argv, "--model")
-    if model_value and model_value != "":
-        # Ensure the last argument is a valid path
-
-        has_delimeter = "=" in argv[model_index]
-        max_args = model_index + (1 if has_delimeter else 2)
-        is_extra_arg_present = len(argv) > max_args
-        is_last_arg_not_option = not argv[-1].startswith("--")
-
-        if is_extra_arg_present and is_last_arg_not_option:
-            path_value = argv[-1]
-            if path_value:
-                return model_value
-            else:
-                logger.error("Missing last path argument. Please provide a valid path.")
-                return None
-        else:
-            logger.error("Missing or invalid last path argument. Please provide a valid path.")
-            return None
-    else:
-        logger.error(
-            "The '--model' argument must be followed by a valid model name, like '--model=gpt4o'.\n"
-            "More details on model options: https://hallux.dev/docs/user-guide/backends"
-        )
-        return None
-
-
 def main(argv: list[str] | None = None, run_path: Path | None = None) -> int:
```

Comment:
Refactor this function to reduce its Cognitive Complexity from 26 to the 15 allowed.
```suggestion
def main(argv: list[str] | None = None, run_path: Path | None = None) -> int:
```